### PR TITLE
Fabo/allow to sign messages for Terra

### PR DESCRIPTION
**Pull request by faboweb**
Architecture for creating messages cross networks

### CODE REVIEW

File: src/ActionModal/utils/MessageConstructor.js [hunk 1]
```diff
@@ -62,174 +56,31 @@ export const getMultiMessage = async (context, messages) => {
...
-      amount: amounts.map(Coin)
-    }
-  }
-}
+async function getMessageFormatter(network, messageType) {
+  let networkMessages
+  try {
+    networkMessages = await import(`./networkMessages/${network}.js`)
+  } catch (err) {
+    throw new Error("Signing for this network is not enabled.")
```
Comment (jbibla) [inline src/ActionModal/utils/MessageConstructor.js]: this error seems like it's for a user - but could a user encounter it? feels like they shouldn't get to the point where we're requesting networkMessage schema if transactions aren't enabled.

File: src/ActionModal/utils/MessageConstructor.js [hunk 2]
```diff
@@ -62,174 +56,31 @@ export const getMultiMessage = async (context, messages) => {
...
-  return {
-    type: `cosmos-sdk/MsgDelegate`,
-    value: {
-      delegator_address: senderAddress,
-      validator_address: validatorAddress,
-      amount: Coin({ amount, denom })
-    }
+  if (!messageFormatter) {
+    throw new Error(
+      `Creating the message ${messageType} for the network ${network} is not supported`
```
Comment (jbibla) [inline src/ActionModal/utils/MessageConstructor.js]:
```suggestion
      `Creating the '${messageType}' message for the ${network} network is not supported`
```

Comment (jbibla) [reply src/ActionModal/utils/MessageConstructor.js]: 😸

Comment (jbibla) [reply src/ActionModal/utils/MessageConstructor.js]: on second thought: these are user error messages so i think they should be helpful and actionable. maybe we could say "there's currently an issue with x transactions on this network. the lunie team is working on a fix. please try again later"

would that make sense?

File: src/ActionModal/utils/networkMessages/terra-testnet.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import * as CosmosMessages from "./cosmos-hub-mainnet.js"
```
Comment (jbibla) [inline src/ActionModal/utils/networkMessages/terra-testnet.js]: should this come from the API?

is `msg.type` the only thing that changed or are the data structures different?

**Comment by jbibla** [comment]
this is awesome. i can feel us _scaling rapidly_ 🚀

File: src/ActionModal/utils/MessageConstructor.js (hunk 1 above)
Comment (faboweb) [reply src/ActionModal/utils/MessageConstructor.js]: True! This is not only for the user but also a good message for us developers. :)

File: src/ActionModal/utils/networkMessages/terra-testnet.js (hunk 3 above)
Comment (faboweb) [reply src/ActionModal/utils/networkMessages/terra-testnet.js]: I don't know. Maybe there could be structural differences down the line. I didn't even check all Terra transactions yet.

File: src/ActionModal/utils/MessageConstructor.js (hunk 2 above)
Comment (faboweb) [reply src/ActionModal/utils/MessageConstructor.js]: Same here: We should prevent this from ever surfacing to the user. The error is a backup

File: src/ActionModal/components/SendModal.vue [hunk 4]
```diff
@@ -271,20 +275,6 @@ export default {
         )
       }
     },
-    setTokenAndBalance() {
```
Comment (jbibla) [inline src/ActionModal/components/SendModal.vue]: isn't this needed for multi denom support?

Comment (faboweb) [reply src/ActionModal/components/SendModal.vue]: this is a computed now

**Pull request merged**